File: car-lease-service/src/test/java/org/jesperancinha/car/lease/repository/CustomerRepositoryTest.java
package org.jesperancinha.car.lease.repository;

import org.jesperancinha.car.lease.model.Customer;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
class CustomerRepositoryTest {

    @Autowired
    private CustomerRepository customerRepository;

    @Test
    public void testSaveCar_whenGoodCar_thenSaveCar() {
        final var customer = Customer.builder().name("customer").build();
        final var customerSave = customerRepository.save(customer);
        assertThat(customerSave.getId()).isNotNull();
        assertThat(customerSave.getName()).isEqualTo("customer");
    }
}

File: car-lease-service/src/test/kotlin/org/jesperancinha/car/lease/repository/CustomerRepositoryTest.kt
package org.jesperancinha.car.lease.repository

import org.jesperancinha.car.lease.model.Customer
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest

@DataJpaTest
internal class CustomerRepositoryTest {
    @Autowired
    private val customerRepository: CustomerRepository? = null
    @Test
    fun testSaveCar_whenGoodCar_thenSaveCar() {
        val customer: Unit = Customer.builder().name("customer").build()
        val customerSave = customerRepository!!.save<Customer>(customer)
        assertThat(customerSave.getId()).isNotNull()
        assertThat(customerSave.getName()).isEqualTo("customer")
    }
}